src/Features/CSharp/Portable/Completion/KeywordRecommenders/ExternKeywordRecommender.cs

```diff
@@ -51,7 +52,7 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
             var syntaxTree = context.SyntaxTree;
             return
                 IsExternAliasContext(context) ||
-                context.IsGlobalStatementContext ||
+                (context.IsGlobalStatementContext && syntaxTree.IsScript()) ||
```

Comment:
do all usages of IsGlobalStatementContext end up needing to distinguish between Script/Non-script?  If so, consider IsGlobalScriptStatementContext and IsGlobalSimpleProgramStatementContext.

Reply:
No, most do not need to distinguish.